trpcultivate_phenotypes/tests/src/Kernel/Validators/MetadataInputTest.php

```diff
@@ -0,0 +1,180 @@
+    $this->test_genus['not-configured'] = $genus;
+    $config = \Drupal::configFactory()->getEditable('trpcultivate_phenotypes.settings')
+      ->set('trpcultivate.phenotypes.ontology.cvdbon.' . $this->test_genus['not-configured'], ['trait' => 0]);
+    
+    // Set terms configuration.
+    $this->setTermConfig();
+  }
+
+  /**
+   * Test genus input.
+   */
+  public function testGenusInput() {
+    // Create a plugin instance for this validator
+    $validator_id = 'trpcultivate_phenotypes_validator_genus_exists';
+    $instance = $this->plugin_manager->createInstance($validator_id);
+    
+    // The validator expects the form values entered into the form elements
+    // through the $form_state.
+    
+    // Create form fields and set value.
+    // This would have been a form that has been submitted.
+    $form_state = new FormState();
+    // A random field.
+    $form_state->setValues(['project' => uniqid()]);
+
+    // Test items that will throw exception:
+    // 1. Passing object or the entire $form_state.
+    // 2. Failed to implement a form field element with genus name/key.
+
+    // Test passing the $form_state.
+    $exception_caught  = FALSE;        
+    try {
+      $instance->validateMetadata($form_state);
+    }
+    catch (\Exception $e) {
+      $exception_caught  = TRUE;
+    }
+    
+    $this->assertTrue($exception_caught, 'Failed to catch exception when passing a $form_state to genus metadata validator.');
+    $this->assertStringContainsString('Unexpected object type was passed', $e->getMessage(), 
+      'Expected exception message does not match message when passing $form_state to genus metadata validator.');
+    
+    
+    // No genus field.
+    $form_values = $form_state->getValues();
+
+    $exception_caught  = FALSE;        
+    try {
+      $instance->validateMetadata($form_values);
+    }
+    catch (\Exception $e) {
+      $exception_caught  = TRUE;
+    }
+
+    $this->assertTrue($exception_caught, 'Failed to catch exception when no genus form field was implemented.');
+    $this->assertStringContainsString('Failed to locate genus field element', $e->getMessage(), 
+      'Expected exception message does not match message when importer failed to implement a form field element with the name/key genus.');
+
+    // Other tests:
+    // Each test will test that genusExists generated the correct title, valid status and failed item.
+    // Failed item is the failed genus value. Additional failed information is attached
+    // whether the genus failed because it does not exist or not configured.
+
+    // Test passing $form_state values with a genus field but genus does not exits.
+    $genus = 'genus-' . uniqid();
+    $form_state->setValues(['genus' => $genus]);
+    $form_values = $form_state->getValues();  
+    $validation_status = $instance->validateMetadata($form_values);
+    
+    $this->assertEquals('Genus exists and is configured with phenotypes', $validation_status['case'],
```

Comment:
Once the changes are made to genusExists(), each of these assertEquals can now check for unique strings that indicate which case occurred